File: src/main/scala/com/dolly/wc/WordCount.scala
package com.dolly.wc

import org.apache.flink.api.scala._

/**
 * 批处理word count程序
 * @author yusenyang 
 * @create 2020/8/1 18:06
 */
object WordCount {

  def main(args: Array[String]): Unit = {
    // 创建一个执行环境
    val env = ExecutionEnvironment.getExecutionEnvironment

    // 从文件中读取数据
    val inputPath = "/Users/sgcx017/github/flink-tutorial/src/main/resources/hello.txt"
    val inputDataSet = env.readTextFile(inputPath)

    // 切分数据得到word，然后再按word做分组聚合
    val wordCountDataSet = inputDataSet
      .flatMap(_.split(" "))
      .map((_,1))
      .groupBy(0)
      .sum(1)

    // 打印结果
    wordCountDataSet.print()
  }
}

File: src/main/java/com/github/dolly0526/flink/demo/WordCount.java
package com.github.dolly0526.flink.demo;

import org.apache.flink.api.java.DataSet;
import org.apache.flink.api.java.ExecutionEnvironment;
import org.apache.flink.api.java.tuple.Tuple2;

/**
 * @author yusenyang
 * @create 2021/3/16 14:25
 */
public class WordCount {

    public static void main(String[] args) throws Exception {

        // 创建执行环境
        ExecutionEnvironment env = ExecutionEnvironment.getExecutionEnvironment();

        // 从文件中读取数据
        String inputPath = "/Users/sgcx017/github/flink-demo/flink-tutorial/src/main/resources/hello.txt";
        DataSet<String> inputDataSet = env.readTextFile(inputPath);

        // 空格分词打散之后，对单词进行 groupby 分组，然后用 sum 进行聚合
        DataSet<Tuple2<String, Integer>> wordCountDataSet = inputDataSet
                .flatMap(new MyFlatMapper())
                .groupBy(0)
                .sum(1);

        // 打印输出
        wordCountDataSet.print();
    }
}
